cmd/monaco/deploy/deploy.go

```diff
@@ -417,6 +402,32 @@ func toProjectMap(projects []project.Project) map[string]project.Project {
 	return result
 }
 
-func containsName(names []string, name string) bool {
-	return slices.Contains(names, name)
+func checkEnvironments(configs project.ConfigsPerEnvironment, envs manifest.Environments) error {
+	for envName, cfgs := range configs {
+		if _, found := envs[envName]; !found {
+			return fmt.Errorf("cannot find environment `%s`", envName)
+		}
+		if err := checkConfigsForEnvironment(envs[envName], cfgs); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func checkConfigsForEnvironment(env manifest.EnvironmentDefinition, cfgs []config.Config) error {
+	for i := range cfgs {
+		if onlyAvailableOnPlatform(&cfgs[i]) && !platformEnvironment(env) && !cfgs[i].Skip {
```

Comment:
advanced nit: assuming that the "onlyAvailableOnPlatform" check will grow in complexity in the future, you could lead with the `!cfgs[i].Skip` and save on the more costly checks if the config is skipped anyway